File: internal/analysis/testdata/callables.go
package conformance

func Outer(flag bool) int {
	handler := func(value int) int {
		if value > 0 && flag {
			return value
		}
		return 0
	}
	if flag {
		return handler(1)
	}
	return 0
}

var PackageHandler = func(values []int) int {
	for range values {
	}
	return len(values)
}

func Register() {
	Use(func(value int) bool {
		return value > 0 || value < -10
	})
}

File: internal/analysis/testdata/callables.rs
pub struct Worker;

impl Worker {
    pub fn outer(&self, flag: bool) -> i32 {
        let handler = |value: i32| {
            if value > 0 && flag {
                return value;
            }
            0
        };
        if flag {
            return handler(1);
        }
        0
    }
}

mod registry {
    pub fn register(values: &[i32]) -> usize {
        values.iter().filter(|value| **value > 0 || **value < -10).count()
    }
}
